build_runner_core/lib/src/environment/create_merged_dir.dart

```diff
@@ -88,57 +88,68 @@ Future<bool> _createMergedOutputDir(
+    if (root != '' &&
+        !builtAssets
+            .where((id) => id.package == packageGraph.root.name)
+            .any((id) => p.isWithin(root, id.path))) {
+      _logger.severe('No assets exist in $root, skipping output');
+      return false;
+    }
 
-  var outputAssets = <AssetId>[];
+    var outputAssets = <AssetId>[];
 
-  await logTimedAsync(_logger, 'Creating merged output dir `$outputPath`',
-      () async {
-    if (!outputDirExists) {
-      await outputDir.create(recursive: true);
-    }
+    await logTimedAsync(_logger, 'Creating merged output dir `$outputPath`',
+        () async {
+      if (!outputDirExists) {
+        await outputDir.create(recursive: true);
+      }
 
-    outputAssets.addAll(await Future.wait(builtAssets.map((id) => _writeAsset(
-        id, outputDir, root, packageGraph, reader, symlinkOnly, hoist))));
+      outputAssets.addAll(await Future.wait(builtAssets.map((id) => _writeAsset(
+          id, outputDir, root, packageGraph, reader, symlinkOnly, hoist))));
 
-    var packagesFileContent = packageGraph.allPackages.keys
-        .map((p) => '$p:packages/$p/')
-        .join('\r\n');
-    var packagesAsset = AssetId(packageGraph.root.name, '.packages');
-    await _writeAsString(outputDir, packagesAsset, packagesFileContent);
-    outputAssets.add(packagesAsset);
+      var packagesFileContent = packageGraph.allPackages.keys
+          .map((p) => '$p:packages/$p/')
+          .join('\r\n');
+      var packagesAsset = AssetId(packageGraph.root.name, '.packages');
+      await _writeAsString(outputDir, packagesAsset, packagesFileContent);
+      outputAssets.add(packagesAsset);
 
-    if (!hoist) {
-      for (var dir in _findRootDirs(builtAssets, outputPath)) {
-        var link = Link(p.join(outputDir.path, dir, 'packages'));
-        if (!link.existsSync()) {
-          link.createSync(p.join('..', 'packages'), recursive: true);
+      if (!hoist) {
+        for (var dir in _findRootDirs(builtAssets, outputPath)) {
+          var link = Link(p.join(outputDir.path, dir, 'packages'));
+          if (!link.existsSync()) {
+            link.createSync(p.join('..', 'packages'), recursive: true);
+          }
         }
       }
-    }
-  });
+    });
 
-  await logTimedAsync(_logger, 'Writing asset manifest', () async {
-    var paths = outputAssets.map((id) => id.path).toList()..sort();
-    var content = paths.join(_manifestSeparator);
-    await _writeAsString(
-        outputDir, AssetId(packageGraph.root.name, _manifestName), content);
-  });
+    await logTimedAsync(_logger, 'Writing asset manifest', () async {
+      var paths = outputAssets.map((id) => id.path).toList()..sort();
+      var content = paths.join(_manifestSeparator);
+      await _writeAsString(
+          outputDir, AssetId(packageGraph.root.name, _manifestName), content);
+    });
 
-  return true;
+    return true;
+  } on FileSystemException catch (e) {
+    if (e.osError.errorCode != 1314) rethrow;
```

Comment:
None of these members can be null correct?

Reply:
good catch, https://api.dartlang.org/stable/2.3.1/dart-io/FileSystemException/osError.html

if only the language had a way to express that at the type system level... :D